.circleci/config.yml

```diff
@@ -0,0 +1,44 @@
+version: 2.1
+orbs:
+  aws-cli: circleci/aws-cli@1.3.2
+  aws-ecr: circleci/aws-ecr@6.15.2
+
+commands:
+
+  create_image_tag:
+    description: Create image tag based on the current branch
+    steps:
+      - run:
+          name: Create IMAGE_TAG env var based on the current branch
+          command: |
+            if [[ -n "${CIRCLE_TAG:-}" ]]; then
+              image_tag=${CIRCLE_TAG}
+            elif [[ $CIRCLE_BRANCH = "master" || $CIRCLE_BRANCH = "staging" ]]; then
+              image_tag=${CIRCLE_BRANCH}-${CIRCLE_SHA1}
+            else
+              image_tag=branch-$(echo $CIRCLE_BRANCH | tr / -)
+            fi
+            echo "export IMAGE_TAG='${image_tag}'" >> $BASH_ENV
+
+jobs:
+  release:
+    machine:
+      docker_layer_caching: true
+    steps:
+      - aws-cli/install
+      - checkout
+      - create_image_tag
+      - aws-ecr/build-and-push-image:
+          dockerfile: Dockerfile
+          region: AWS_DEFAULT_REGION
+          repo: onemedical/aws-es-proxy
+          tag: ${IMAGE_TAG}
+
+workflows:
+  version: 2
+  everything:
+    jobs:
+      - release:
+          filters:
+            tags:
+              only: /^v\d+\.\d+\.\d+/
```

Comment:
So this will use the version tag for builds?

Reply:
Not quite. There's no restriction on branch builds. This filter makes it so that only tags that match that version number regex pattern will build on circle. All other branches will build normally

Reply:
As you know lean metrics expect release tags in the format release-##. 
It doesn't currently support releases tag with that regex pattern

Reply:
@dgeorges this is for a sidecar process, which complicates the release tags, for a lot of reasons. I think that belongs pretty high up on our debt board today, but it needs some work.

Reply:
yep, that would complicate collecting the deploy metrics, which maybe doesn't make much sense for side cars. but we can still collect release velocity information. I can add this pattern to be supported.

Reply:
We should be versioning sidecar containers, it just makes maintenance complicated, since we don't manage task defs well.